src/geo.rs

```diff
@@ -98,22 +101,22 @@ pub fn subdivide_triangle(triangle: Triangle) -> Vec<Triangle> {
         return vec![triangle];
     }
 
-    let d01 = largest_angle(c0, c1);
-    let d12 = largest_angle(c1, c2);
-    let d20 = largest_angle(c2, c0);
+    let c0c1_error = error(c0, c1);
+    let c1c2_error = error(c1, c2);
+    let c2c0_error = error(c2, c0);
 
     let mut too_long_side = false;
-    if d01 > MAX_COORDINATE_ANGLE_RAD_TRIANGLES {
+    if c0c1_error > MAX_TRIANGLE_CUT_THROUGH_SPHERE_ERROR {
```

Comment:
Just a small remark:

The largest cut through of a triangle might not be the center of either edge, it might be the center point of the triangle; e.g in the case that all edges are of equal length. Thus all these conditions below could be false, but we are still having a point in on the triangle that have more cut through error.

I don't really know if we need to care about it though, I'll leave it up to you to decide if we want to act on it.

Reply:
This is an interesting thought. And the algorithm was always intended to be an approximation. If we indeed would compute the error at the point in the triangle furthest from the earth's surface, then it would be an exact algorithm. Then we could guarantee it would never end up below the ocean etc.

I leave that up for a future exercise.